### PR TITLE
Improve display of ROR information in deposit form

### LINKED ISSUE #1805: Improve display of ROR information in deposit form

### PR DESCRIPTION
**Pull request by carlinmack**
Close #1805

Relies on https://github.com/inveniosoftware/react-invenio-forms/pull/255

:heart: Thank you for your contribution!

### Description

> Please describe briefly your pull request.

(I don't know why the dropdown boxes are so small in the screenshots)

In person view I fixed the margin above "Role", removed the extra space to the left of the ORCID logo, and added the affiliations. 

Before and after:

image

In the organisation view I removed the ROR icon which shouldn't be there (according to ROR) and added the new information we have added to the API

Before and after:

image

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js [hunk 1]
```diff
@@ -246,10 +246,9 @@ export class CreatibutorsModal extends Component {
       icon = "/static/images/gnd-icon.svg";
       link = "https://d-nb.info/gnd/" + identifier.identifier;
     } else if (identifier.scheme === "ror") {
-      icon = "/static/images/ror-icon.svg";
-      link = "https://ror.org/" + identifier.identifier;
+      return; // ROR doesn't recommend displaying ROR IDs
```
[the remark this answers is not in the document]
Comment (carlinmack) [reply invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js:249]: And yep changing to a mapping sounds good!

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js [hunk 2]
```diff
@@ -261,38 +260,51 @@ export class CreatibutorsModal extends Component {
...
           {identifier.scheme === "orcid" ? identifier.identifier : null}
         </a>
       </span>
     );
   };
 
+  makeSubheader = (creatibutor) => {
+    const { isOrganization } = this.state;
+
+    if (isOrganization) {
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js:273]: there is a chance to add this in react-invenio-forms:

https://github.com/inveniosoftware/react-invenio-app-ils/blob/master/src/lib/components/SeparatedList/SeparatedList.js and use it for displaying the comma separated lists without replicating the code below

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js [hunk 3]
```diff
@@ -261,38 +260,51 @@ export class CreatibutorsModal extends Component {
...
-        creatibutor.affiliations.forEach((affiliation, idx) => {
-          affNames += affiliation.name;
-          if (idx < creatibutor.affiliations.length - 1) {
-            affNames += ", ";
-          }
-        });
-      }
+      const subheader = this.makeSubheader(creatibutor);
+      let name = creatibutor.name;
+      if (creatibutor.acronym) name += " (" + creatibutor.acronym + ")";
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js:301]:
```suggestion
      if (creatibutor.acronym) name += `(${creatibutor.acronym})`;
```

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js [hunk 4]
```diff
@@ -301,10 +313,8 @@ export class CreatibutorsModal extends Component {
         key: creatibutor.id,
         content: (
           <Header>
-            {creatibutor.name} {idString.length ? <>({idString})</> : null}
-            <Header.Subheader>
-              {isOrganization ? creatibutor.acronym : affNames}
-            </Header.Subheader>
+            {name} {idString.length ? <>({idString})</> : null}
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js:316]:
```suggestion
            {name} {idString.length && <>({idString})</>}
```

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js [hunk 5]
```diff
@@ -235,108 +234,14 @@ export class CreatibutorsModal extends Component {
...
-            <Header.Subheader>
-              {isOrganization ? creatibutor.acronym : affNames}
-            </Header.Subheader>
-          </Header>
-        ),
-      };
-    });
-
-    const { showPersonForm } = this.state;
+  SuggestionsWrapper = (creatibutors) => {
```
Comment (zzacharo) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js:237]: we typically use this name convention for classes. Shoulnd't that be just a serialize method that returns the `AffiliationsSuggestions`?

Comment (carlinmack) [reply invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js:237]: yep! the naming is still WIP will change

**Review by kpsherva — APPROVED** [review]

**Pull request merged**